streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollector.java

```diff
@@ -25,7 +25,7 @@
 
 import java.util.Map;
 
-public interface RecordCollector extends AutoCloseable {
```

Comment:
We never used it as an AutoCloseable, and having it makes it hard to trace the callers.